File: FitnessApi/FitnessApi/Models/WorkoutExercise.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace FitnessApi.Models;

public partial class WorkoutExercise
{
    public int WorkoutId { get; set; }

    public int ExerciseId { get; set; }

    public int? Sets { get; set; }

    public int? Reps { get; set; }

    public float? WeightKg { get; set; }
    [JsonIgnore]
    public virtual Exercise Exercise { get; set; } = null!;
    [JsonIgnore]
    public virtual Workout Workout { get; set; } = null!;
}

File: FitnessApp/app/src/main/java/com/example/fitnessapp/data/model/WorkoutExercise.kt
package com.example.fitnessapp.data.model

data class WorkoutExercise(
    val workoutId: Int,
    val exerciseId: Int,
    val sets: Int?,
    val reps: Int?,
    val weightKg: Float?,
    val previewImageUrl: String? = null
)


